### PR TITLE
[FEATURE] Add `DataAssetStore`

### PR DESCRIPTION
**Pull request by roblim**
- Add DataAssetStore, to be used only by CloudDataContexts, to enable OSS to hit new `data-assets` cloud endpoint
- Update Datasource.delete_asset to use above endpoint to delete assets directly, instead of through Datasource updates

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/data_context/data_context/cloud_data_context.py [hunk 1]
```diff
@@ -439,6 +445,40 @@ def _init_datasource_store(self) -> DatasourceStore:
         )
         return datasource_store
 
+    def _init_data_asset_store(self) -> DataAssetStore:
+        from great_expectations.data_context.store import DataAssetStore
+        from great_expectations.data_context.store.gx_cloud_store_backend import (
+            GXCloudStoreBackend,
+        )
```
Comment (cdkini) [inline great_expectations/data_context/data_context/cloud_data_context.py]: Cyclic import issues?

File: great_expectations/data_context/data_context/cloud_data_context.py [hunk 2]
```diff
@@ -439,6 +445,40 @@ def _init_datasource_store(self) -> DatasourceStore:
...
+
+        # Never explicitly referenced but adheres
+        # to the convention set by other internal Stores
+        store_name = DataContextConfigDefaults.DEFAULT_DATA_ASSET_STORE_NAME.value
+        store_backend: dict = {"class_name": GXCloudStoreBackend.__name__}
+        runtime_environment: dict = {
+            "root_directory": self.root_directory,
+            "ge_cloud_credentials": self.ge_cloud_config.to_dict(),  # type: ignore[union-attr]
+            "ge_cloud_resource_type": GXCloudRESTResource.DATA_ASSET,
+            "ge_cloud_base_url": self.ge_cloud_config.base_url,  # type: ignore[union-attr]
```
Comment (cdkini) [inline great_expectations/data_context/data_context/cloud_data_context.py:428]: Appreciate you using enums and `__name__`s 🙇🏽

File: great_expectations/datasource/fluent/interfaces.py [hunk 3]
```diff
@@ -517,10 +517,16 @@ def delete_asset(self, asset_name: str) -> None:
...
+
         asset: _DataAssetT
+        asset = self.get_asset(asset_name=asset_name)
         self.assets = list(filter(lambda asset: asset.name != asset_name, self.assets))
 
-        self._save_context_project_config()
+        if self._data_context and isinstance(self._data_context, CloudDataContext):
+            self._data_context._delete_asset(id=str(asset.id))
+        else:
+            self._save_context_project_config()
```
Comment (cdkini) [inline great_expectations/datasource/fluent/interfaces.py:531]: Hmm thinking out loud but is there a way we can avoid `isinstance` for a Cloud-specific path? Probably not

**Review by cdkini — APPROVED** [review]

File: great_expectations/data_context/store/data_asset_store.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        """
+        if isinstance(value, dict):
+            # presence of a 'type' field means it's a fluent DataAsset
+            type_ = value.get("type")
+            data_asset_model = _SourceFactories.type_lookup.get(type_)
+            if not data_asset_model:
+                raise LookupError(f"Unknown DataAsset 'type': '{type_}'")
+            return data_asset_model(**value)
+        else:
+            return self._schema.loads(value)
```
Comment (Kilo59) [inline great_expectations/data_context/store/data_asset_store.py]: I think we can simply the code and just assume a fluent asset.

`.parse_raw()` parses a json string. `self._schema.loads()` is using a Marshmallow schema which we don't have for a FluentDataAsset.
```suggestion
        if isinstance(value, dict):
            # presence of a 'type' field means it's a fluent DataAsset
            type_ = value.get("type")
            data_asset_model = _SourceFactories.type_lookup.get(type_)
            if not data_asset_model:
                raise LookupError(f"Unknown DataAsset 'type': '{type_}'")
            return data_asset_model(**value)
       return data_asset_model.parse_raw(value)
```

File: great_expectations/datasource/fluent/interfaces.py (hunk 3 above)
Comment (Kilo59) [reply great_expectations/datasource/fluent/interfaces.py:531]: I think there's an attribute we could check for, but that's essentially the same thing.

File: great_expectations/data_context/store/data_asset_store.py (hunk 4 above)
Comment (roblim) [reply great_expectations/data_context/store/data_asset_store.py]: Ah, good catch.

File: great_expectations/datasource/fluent/interfaces.py (hunk 3 above)
Comment (roblim) [reply great_expectations/datasource/fluent/interfaces.py:531]: I thought there was a property too but didn't find one. @cdkini - can you remind me why we should avoid isinstance where possible?

**Pull request merged**